lesson 1/README.md

```diff
@@ -0,0 +1,363 @@
+    }
+    
+    public void signUpForChecking(int amount) {
+    }
+
+    public static void main(String[] args) {
+    }
+}
+
+```
+
+```
+public class Account {
+
+    private MoneySaver owner;
+    private long amount;
+
+    public String toString() {
+        return "Account owner: " + owner.getName() + ", Account Balance: $" + amount;
+    }
+
+    public static Account largerAccount(Account acc1, Account acc2) {
+    }
+
+}
+
+```
+
+##### Constructors
+
+The constructor is a special method used to initialize instances of your class (just like Python's `__init__` method). Classes by default inherit a no-argument constructor from the default `Object` class. That means that even without epxlicitly defining a constructor, we can already initalize an `Account`:
+
+```Account a = new Account() // This works!```
+
+Often, you will want to initialize your object with some values. In this case you must define your own constructor.
+
+**Go ahead and implement the `Account` constructor. It should have input arguments `long amount, MoneySaver owner`**
+
+#### Fields
+
+You'll notice both classes also have fields. Fields are either attributes of members of a class (an `amount` describes an `Account`) or things that belong to the class (an `Account` has an `owner`). **Fields should almost exclusively be `private` unless you have a *really* good reason to make it `public`.** If you want an outside class to be able to access a field, you must create public getter/setter methods.
+
+**Define and implement the `getAmount` and `setAmount` methods in `Account`. Make sure they have appropriate return types and input arguments.**
+
+Add the following to your main method (`public static void main(String[] args)`) to test your code so far:
+
+```
+MoneySaver jim = new MoneySaver("Jim", 100);
+Account a = new Account(100, jim);
+System.out.println(a);
+a.setAmount(20);
+System.out.println("New amount: " + a.getAmount());
+```
+
+Expected output:
+
+```
+Account owner: Jim, Account Balance: $100
+New amount: 20
+```
+
+#### Methods
+
+These classes also have methods, which are things instances can "do". Each method must specify what it returns, (`void` if nothing). Generally speaking methods are `public`. However, you might find yourself writing a `private` method that nothing outside the class should be able to see. Often, "helper" methods are `private`.
+
+**implement `deposit` in `Account`.**
+
+Test your code:
+
+```
+MoneySaver jim = new MoneySaver("Jim", 100);
+Account a = new Account(100, jim);
+System.out.println(a);
+a.setAmount(20);
+a.deposit(10);
+System.out.println("New amount: " + a.getAmount());
+```
+
+```
+Account owner: Jim, Account Balance: $100
+New amount: 30
+```
+
+##### What does `static` mean?
+
+`static` methods operate on the *Class*, as opposed to *instances* of the class. You can call static methods without creating any object e.g. `Collections.sort()`. Make a method `static` if the method is something the *Class* can do.
+
```

Comment:
domething the 'Class' can do **and doesn't need any instance variables to work properly**